drivers/hmis/spec/requests/hmis/client_alert_spec.rb

```diff
@@ -35,6 +35,23 @@
     GRAPHQL
   end
 
+  let(:create_alert) do
+    <<~GRAPHQL
+      mutation CreateClientAlert($id: ID!, $note: String!, $expirationDate: ISO8601Date, $priority: ClientAlertPriorityLevel) {
```

Comment:
@gigxz the way `expirationDate` needed to be in camel case here really baffled me. When I had it in snake case, graphql didn't fail loudly, but the query would simply silently fail to have an expiration date by the time it got to the mutation ruby code (at which point it's in snake case anyway). How can I better understand where and why graphql does this weird magic with changing cases around?

Reply:
On this line, `$expirationDate` is just a variable name, it could be `$foo`. As long as you pass it to the `expirationDate` argument on the line below. The argument casing is `expirationDate` because graphql uses camel case. Ruby-graphql handles the camelizing of the Ruby-defined arguments and field names.

More on variables: https://graphql.org/learn/queries/#:~:text=.-,Variables

Reply:
thank you!